### PR TITLE
jasmine: add a few new types

### PR DESCRIPTION
**Pull request by UziTech**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/jasmine/jasmine>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by UziTech** [comment]
TODO:
- [ ] Rebase on #51997 after it is merged
- [ ] Add tests

**Comment by chivesrs** [comment]
Just make sure you're not duplicating work that's being done by one of the Jasmine core maintainers in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51998

**Comment by sgravrock** [comment]
Thanks for all your work to bring things up to date. Work like this takes a huge burden off the Jasmine maintainer, and we appreciate it.

Some of the additions in this and a couple of your other current PRs are private APIs. I wanted to mention `PrettyPrinter` in particular because it's private, it's undergone significant churn lately, and it's likely to change in a future Jasmine 3.x release if some of my plans work out. In general, anything that's not documented at https://jasmine.github.io/ is a private API that could be changed or removed from Jasmine on a future point release. I'd prefer that those not be included in the type definitions.

The exception is that there are a few things that should be public but aren't in the docs. For instance, we probably need to make at least a bit of `Spec` public or else [Configuration#specFilter](https://jasmine.github.io/api/edge/Configuration.html) isn't useful). If you run into anything like that I'd appreciate an issue or PR on the main Jasmine project or docs project.

**Comment by sgravrock** [comment]
Looks good to me.

File: types/jasmine/jasmine-tests.ts [hunk 1]
```diff
@@ -1355,6 +1355,80 @@ describe("jasmine.arrayWithExactContents", () => {
     });
 });
 
+describe("jasmine.mapContaining", () => {
```
Comment (chivesrs) [inline types/jasmine/jasmine-tests.ts:1358]: These tests fail at runtime. Any chance you can give me things that work at runtime so I can verify the types are right?

**Review by chivesrs — CHANGES_REQUESTED** [review]

File: types/jasmine/jasmine-tests.ts (hunk 1 above)
Comment (chivesrs) [reply types/jasmine/jasmine-tests.ts:1358]: image

**Comment by UziTech** [comment]
It should pass now. It was failing because AsymmetricMatchers don't work with `toBe`. I just copied the `arrayContaining` tests which should fail the same way. Do you want me to update the `arrayContaining` tests so they pass as well?

**Review by chivesrs — APPROVED** [review]

**Comment by chivesrs** [comment]
Nah leave this PR as its own atomic unit. If you want to update arrayContaining, then file a new PR. I'm really only concerned with the diff of this PR, not the state of the tests as a whole.

**Comment by UziTech** [comment]
Ready to merge

**Pull request merged**